ramp-database/rampdb/model/event.py

```diff
@@ -76,110 +159,265 @@ class Event(Model):
+    def __init__(self, problem_name, name, event_title, session=None):
         self.name = name
-        # to check if the module and all required fields are there
-        # db fields are later initialized by tools._set_table_attribute
-        self.problem = Problem.query.filter_by(name=problem_name).one()
+        if session is None:
+            self.problem = Problem.query.filter_by(name=problem_name).one()
+        else:
+            self.problem = (session.query(Problem)
+                                   .filter(Problem.name == problem_name)
+                                   .one())
         self.title = event_title
-        self.Predictions
 
     def __repr__(self):
-        repr = 'Event({})'.format(self.name)
-        return repr
+        return 'Event({})'.format(self.name)
 
     def set_n_submissions(self):
+        """Set the number of submissions for the current event by checking
+        each team."""
         self.n_submissions = 0
         for event_team in self.event_teams:
             # substract one for starting kit
             self.n_submissions += len(event_team.submissions) - 1
 
     @property
     def Predictions(self):
+        """:class:`rampwf.prediction_types.base.BasePrediction`: Predictions
+        for the given event."""
         return self.problem.Predictions
 
     @property
     def workflow(self):
+        """:class:`rampdb.model.Workflow`: The workflow used for the event."""
         return self.problem.workflow
 
     @property
+    # XXX: This will only work with Flask
     def official_score_type(self):
-        return EventScoreType.query.filter_by(
-            event=self, name=self.official_score_name).one()
+        """:class:`rampdb.model.EventScoreType`: The score type for the current
+        event."""
+        return (EventScoreType.query
+                              .filter_by(event=self,
+                                         name=self.official_score_name)
+                              .one())
+
+    def get_official_score_type(self, session):
+        """Get the type of the default score used for the current event.
+
+        Parameters
+        ----------
+        session : :class:`sqlalchemy.orm.Session`
+            The session used to make the query.
+
+        Returns
+        -------
+        event_type_score : :class:`rampdb.model.EventTypeScore`
+            The default type score for the current event.
+        """
+        return (session.query(EventScoreType)
+                       .filter(EventScoreType.event == self)
+                       .filter(EventScoreType.name == self.official_score_name)
+                       .one())
 
     @property
+    # XXX: This will only work with Flask
     def official_score_function(self):
+        """callable: The default function used for scoring in the event."""
         return self.official_score_type.score_function
 
+    def get_official_score_function(self, session):
```

Comment:
Are those additional functions actually used? (outside of the tests)

It is rather duplicative, and would be nice if we could avoid that